### PR TITLE
HPCC-25736 Refactor React Fields, Groups and Forms

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/forms/Forms.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        setLocalFields(localFields);
+        onReset(fieldsToRequest(localFields));
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [doReset]);
+
+    return <TableGroup fields={localFields} onChange={(id, value) => {
+        const field = localFields[id];
+        switch (field.type) {
+            case "checkbox":
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/Forms.tsx:52]: There's not currently any difference between this and the default. Was there something else intended to go here?

File: esp/src/src-react/components/forms/Fields.tsx [hunk 2]
```diff
@@ -100,35 +192,61 @@ type Field = StringField | CheckboxField | DateTimeField | LinkField |
...
+export interface TargetGroupTextFieldProps {
+    key: string;
+    label?: string;
+    selectedKey?: string;
+    className?: string;
+    onChange?: (event: React.FormEvent<HTMLDivElement>, option?: IDropdownOption, index?: number) => void;
+    placeholder?: string;
+}
+
+const TargetGroupTextField: React.FunctionComponent<TargetGroupTextFieldProps> = ({
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/Fields.tsx:242]: Should this have been exported like the others?

**Review by jeclrsg — DISMISSED** [review]
@GordonSmith Had questions on a couple of small things, but looks good to me.

**Review by jbrundage — DISMISSED** [review]

File: esp/src/src-react/components/forms/Forms.tsx (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src-react/components/forms/Forms.tsx:52]: I think there was special code when the checkbox was a material ui checkbox.  Will remove

File: esp/src/src-react/components/forms/Fields.tsx (hunk 2 above)
Comment (GordonSmith) [reply esp/src/src-react/components/forms/Fields.tsx:242]: Probably...

**Comment by GordonSmith** [comment]
@richardkchapman changes made as per above - should be good to pull.

File: esp/src/src-react/components/forms/Forms.tsx (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src-react/components/forms/Forms.tsx]: Removed redundent code.

File: esp/src/src-react/components/forms/Fields.tsx (hunk 2 above)
Comment (GordonSmith) [reply esp/src/src-react/components/forms/Fields.tsx]: Added missing export

**Pull request merged**